pkg/initrequest/event.go

```diff
@@ -0,0 +1,307 @@
+// This file is part of MinIO DirectPV
+// Copyright (c) 2021, 2022 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package initrequest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	directpvtypes "github.com/minio/directpv/pkg/apis/directpv.min.io/types"
+	"github.com/minio/directpv/pkg/client"
+	"github.com/minio/directpv/pkg/consts"
+	pkgdevice "github.com/minio/directpv/pkg/device"
+	"github.com/minio/directpv/pkg/listener"
+	"github.com/minio/directpv/pkg/sys"
+	"github.com/minio/directpv/pkg/types"
+	"github.com/minio/directpv/pkg/utils"
+	"github.com/minio/directpv/pkg/xfs"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+)
+
+const (
+	initReqSpan = time.Hour * 24
```

Comment:
will reduce this to `5mins`. `1 day` doesn't make sense and it is not required. We can cleanup any stale initrequests every 5mins.

Users can trigger the init again to see the failures. By this way, we can avoid too many initrequests getting created.